api/src/api/emailCheckRouter.ts

```diff
@@ -0,0 +1,34 @@
+import { UserDataClient } from "@domain/types";
+import type { LogWriterType } from "@libs/logWriter";
+import { Router } from "express";
+import { StatusCodes } from "http-status-codes";
+
+export const emailCheckRouterFactory = (userDataClient: UserDataClient, logWriter: LogWriterType): Router => {
+  const router = Router();
+
+  router.post("/emailCheck", async (req, res) => {
+    const { email } = req.body;
+
+    const logId = logWriter.GetId();
+    let status;
+
+    if (email === undefined) {
+      status = StatusCodes.BAD_REQUEST;
+      res.status(status).send({ error: "`email` property required." });
+    } else {
+      const userData = await userDataClient.findByEmail(email);
```

Comment:
[boulder] I think it's unlikely, but this logic may trigger some false negative cases. I just looked into the `findByEmail` because I'm not super familiar and it looks like if there is a match on an email address, it will return userData. IF there are multiple matches, it will return undefined.

Based on your current logic, an email address associated with multiple accounts will return undefined and the endpoint will return a 404.

I think we're either going to need to wrap this call in a try/catch or chain then/catch calls. Not super partial on the solution, but I do think this is an issue worth addressing.

Reply:
The key piece is that we need to be looking for a resolved promise not necessarily checking for userData being undefined.

Reply:
@jphechter Is this opportunity for a new `findAllByEmail` method?

Reply:
I think @alexander-littleton's suggestion is worth considering. Right now, though, it looks like `findByEmail` isn't currently being used anywhere else. I'm guessing it either got made as a utility or was used somewhere previously and then the call got removed without removing the function.

Given that, I think it's ok to update this to return the first email match if multiple users are found. To @jphechter's point about it being unlikely, I'm guessing this is more of a testing environment occurrence rather than a prod problem.